silkworm/core/types/account.cpp

```diff
@@ -148,6 +148,55 @@ tl::expected<Account, DecodingError> Account::from_encoded_storage(ByteView enco
     return a;
 }
 
+tl::expected<Account, DecodingError> Account::from_encoded_storage_v3(ByteView encoded_payload) noexcept {
+    Account a;
+    if (encoded_payload.empty()) {
+        return a;
+    }
+    size_t pos{0};
+    for (int i{0}; i < 4; ++i) {
+        uint8_t len = encoded_payload[pos++];
+        if (len == 0) {
+            if (encoded_payload.length() == pos && i < 3) {
+                return tl::unexpected{DecodingError::kUnexpectedLength};
+            }
+            continue;
+        }
+        if (encoded_payload.length() < pos + len) {
+            return tl::unexpected{DecodingError::kInputTooShort};
+        }
+        const auto encoded_value{encoded_payload.substr(pos, len)};
+        switch (i) {
```

Comment:
this loop with a switch is a smart hack to structure the code
it took me a while to figure it out

a more basic way would be to extract this preambule into a helper function or a local lambda,
and then call it before each decoding attempt

Reply:
I did it this way for similarity with the code already present in `Account::from_encoded_storage`